### ISSUE TITLE
nix-channel gets random quotes added after nixos-rebuild 

### ISSUE THREAD

**Issue by pabloaul** [NixOS/nix#9264]
**Describe the bug**
After a rebuild random quotes in the nix-channel list started appearing causing subsequent build to fail.

**Steps To Reproduce**
1. be on channel nixos-unstable and check nix-channel --list to see good configuration
2. do nixos-rebuild switch
3. check nix-channel --list and see broken entries and quote before link after rebuild

**Expected behavior**
a rebuild shouldn't break my nix-channel list

**`nix-env --version` output**
nix-env (Nix) 2.17.1

**Additional context**
![image](https://github.com/NixOS/nix/assets/35423980/e74bcb62-ba20-4873-885f-eca7c9a88103)

**Priorities**

**Comment by Lilacious** [comment]
Happened to me too. I have 2 channels and both had quotes after a rebuild.
`sudo nix-channel --list`
Output:
`nixos "https://nixos.org/channels/nixos-unstable`
`nixos-23.05 "p://nixos.org/channels/nixos-23.05`

**Comment by keanuk** [comment]
Happened to me too on two different machines, I got the following output:
```
sudo nix-channel --list                                               
" "
nixos "https://nixos.org/channels/nixos-unstable
```

I was able to remove both entries with `sudo nix-channel --remove nixos` and `sudo nix-channel --remove \"`
And I added the channel I wanted to use again with `sudo nix-channel --add https://channels.nixos.org/nixos-unstable nixos`
But anytime I run `sudo nixos-rebuild switch` it will add the same broken entries into my nix-channels again.

**Comment by F3l1XQu1ll** [comment]
I'm having the same issues after running `sudo nixos-rebuild switch`.
It seems like this changes the channels list under `/root/.nix-channels` to 

```
"https://nixos.org/channels/nixos-unstable nixos
"
```

even if the name of the channel was different before (tried the default name "nixos" as described in the comments above and different names).

I have set up flakes, my `/etc/nixos/flake.nix` has nixpkgs url set to `nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable"`.
Removing this line does not affect the behaviour.

This installation is also relatively new. I installed NixOS version 23.05 using the latest graphical image a few days ago.
